bindings/python/setup.py

```diff
@@ -82,8 +79,25 @@ def b2_escape(value: str) -> str:
     return f'"{value}"'
 
 
+# Used for understanding a b2 command line. We could get more introspective here,
+# and use python to parse the b2 environment to get all the implicit features
+# from the b2 environment, but this is good enough for now
+IMPLICIT_FEATURE_MAP = {
```

Comment:
this approach worries me a bit. It would be more reliable to require explicit disabling of certain features (I understand why you want to do that now). As long as the defaults are what most people want, I think it's fine to require some additional arguments to get more advanced uses to work.

Reply:
actually, even better, we could require the custom `b2` arguments to always specify the feature name, and not rely on implicit features at all.

Reply:
Sounds good to me, I'll remove this. For `variant` in particular we consider `build_ext --debug` to mean `b2 variant=debug`, so the most-common case is pretty user friendly.